cEP-0025.md

```diff
@@ -0,0 +1,116 @@
+# Integration of pyflakes-enhanced-AST into coala
+
+| Metadata |                                                 |
+| -------- | ----------------------------------------------- |
+| cEP      | 22                                              |
+| Version  | 1.0                                             |
+| Title    | Integration of pyflakes-enhanced-AST into coala |
+| Authors  | Ankit Joshi <mailto:ajankit2304@gmail.com>      |
+| Status   | Proposed                                        |
+| Type     | Feature                                         |
+
+## Abstract
+
+This document describes how to create a metabear to integrate
+pyflakes-enhanced-AST into coala.
+
+## Introduction
+
+The current approach of plugin development for flake8 does not utilises
+the full potential of enhanced AST rather they simply depend on python AST
+and so, a whole lot of rework is required to do the basic traversing
+and collection of important nodes.
+
+Pyflakes provides with a basic API that does the traversing.
+So, if a developer uses enhanced AST he just needs to work on
+the implementation of the new logic that his/her plugin provides
+and not about the fidelity of the basic node handlers.
```

Comment:
and important problem not yet discussed is that pyflakes internals show the state with erroneous objects as-is in the scopes.

That could lead to plugins doing the 'wrong' thing.  Usually the wrong thing has been done in the code, and needs to be fixed in the source, and then coala restarted, so

1. maybe no pyflakes plugins are run unless pyflakes found no errors.
2. erroneous objects are detected (does pyflakes keep enough metadata to do this?), and then either removed (creating more problems?) or flagged (meaning plugin authors need to check a flag..ugh).